src/hostname/hostnamed.c

```diff
@@ -250,8 +250,27 @@ static int get_firmware_vendor(char **ret) {
          return get_hardware_firmware_data("bios_vendor", ret);
 }
 
-static int get_firmware_date(char **ret) {
-         return get_hardware_firmware_data("bios_date", ret);
+static int get_firmware_date(uint64_t *ret) {
+         const char *k;
+         struct tm tm;
+         time_t v;
+         _cleanup_free_ char *bios_date = NULL;
+         int r;
+
+         r = get_hardware_firmware_data("bios_date", &bios_date);
+         if (r <= 0)
+                 return r;
+
+         k = strptime(bios_date, "%m/%d/%Y", &tm);
```

Comment:
I think it is better to use `parse_timestamp()`, though it does not support `/` as a separator. So, how about to first replace `/` with `-`, and use `parse_timestamp()`?